### PR TITLE
Configure retry logic

### PR DESCRIPTION
**Pull request by jdnurme**
Update interface to include configurable values passable from user.

- [x] Tests pass
- [x] Appropriate changes to documentation are included in the PR

### CODE REVIEW

File: dataflux_pytorch/dataflux_iterable_dataset.py [hunk 1]
```diff
@@ -114,15 +129,17 @@ def __iter__(self):
...
+            yield from (
+                self.data_format_fn(bytes_content)
+                for bytes_content in dataflux_core.download.dataflux_download_lazy(
+                    project_name=self.project_name,
+                    bucket_name=self.bucket_name,
+                    objects=self.objects,
+                    storage_client=self.storage_client,
+                    dataflux_download_optimization_params=self.dataflux_download_optimization_params,
+                    retry_config=self.config.download_retry_config,
+                )
```
Comment (bernardhan33) [inline dataflux_pytorch/dataflux_iterable_dataset.py:141]: These reformatting might be coming from a different formatter that you used with the recommended one (checked in by Matt). Could you modify that? This way we can prevent modifying these lines consistently with different formatters.

File: dataflux_pytorch/dataflux_iterable_dataset.py [hunk 2]
```diff
@@ -46,6 +51,12 @@ class Config:
 
         disable_compose: A boolean flag indicating if compose download should be active.
         Compose should be disabled for highly scaled implementations.
+
+        list_retry_config: A google API retry for dataflux fast list operations. This allows
```
Comment (bernardhan33) [inline dataflux_pytorch/dataflux_iterable_dataset.py:55]: Dataflux

File: dataflux_pytorch/dataflux_iterable_dataset.py [hunk 3]
```diff
@@ -46,6 +51,12 @@ class Config:
...
+        for retry backoff configuration.
+
+        download_retry_config: A google API retry for dataflux donwload operations. This allows
```
Comment (bernardhan33) [inline dataflux_pytorch/dataflux_iterable_dataset.py:58]: typo: download

File: README.md [hunk 4]
```diff
@@ -356,6 +356,26 @@ Many machine learning efforts opt for a highly distributed training model levera
 1. Check that other workloads executing within your project are not drawing excess bandwidth. Bandwidth usage can be viewed by following steps [here](https://cloud.google.com/storage/docs/bandwidth-usage#bandwidth-monitoring).
 2. Follow [these instructions](https://cloud.google.com/storage/docs/bandwidth-usage#increase) to apply for a quota increae of up to 1 Tbps.
 3. If more than 1 Tpbs is required, contact your Technical Account Manager or Google representative to file a request on your behalf. This request should specify that you wish to increase the GCS bandwidth caps on your project.
+4. Adjust the `listing_retry_config` and `download_retry_config` options to tune your retry backoff and maximize performance.
+
+    ```python
+    from google.cloud.storage.retry import DEFAULT_RETRY
+
+    dataset = dataflux_mapstyle_dataset.DataFluxMapStyleDataset(
+      project_name=PROJECT_NAME,
```
Comment (bernardhan33) [inline README.md:365]: This python script is indented using 2 spaces. Can we make it consistent with the 4 spaces that we are using throughout the code base?

**Review by bernardhan33 — APPROVED** [review]

**Review by MattIrv — APPROVED** [review]

**Pull request merged**